pallets/roles/src/impls.rs

```diff
@@ -284,18 +284,36 @@ impl<T: Config> sp_runtime::BoundToRuntimeAppPublic for Pallet<T> {
 impl<T: Config> OneSessionHandler<T::AccountId> for Pallet<T> {
 	type Key = T::RoleKeyId;
 
-	fn on_genesis_session<'a, I: 'a>(_validators: I)
+	fn on_genesis_session<'a, I: 'a>(validators: I)
 	where
 		I: Iterator<Item = (&'a T::AccountId, T::RoleKeyId)>,
 	{
-		// nothing to be done
+		validators
+			.into_iter()
+			.filter(|(acc, _)| Ledger::<T>::contains_key(acc))
+			.for_each(|(acc, role_key)| {
```

Comment:
Maybe we should consider queuing this, consider the case with 500 validators, on the session change block this will execute 500 times, might slow down/stall. Easiest here is to make it manual, since `session.rotateKeys` is a manual call, we can just ask validators to also call `roles.updateRoleKey` after that?

Reply:
I guess it is fine in Genesis session? not good on session change.